### PR TITLE
[WFLY-12985] Provide a separate feature pack and dist for non-observability MP specs

### PR DESCRIPTION
**Pull request by bstansberry**
https://issues.redhat.com/browse/WFLY-12985

### CODE REVIEW

**Comment by bstansberry** [comment]
retest this please

**Comment by bstansberry** [comment]
retest this please

**Comment by bstansberry** [comment]
@jamezp I fixed the -Drelease problem. Thanks for pointing it out!

**Review by jamezp — APPROVED** [review]
This looks good to me.

**Comment by bstansberry** [comment]
retest this please

**Comment by bstansberry** [comment]
retest this please

File: ee-galleon-pack/src/main/resources/packages/product.conf/content/modules/system/layers/base/org/jboss/as/product/wildfly-ee/module.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  ~ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  ~ Lesser General Public License for more details.
+  ~
+  ~ You should have received a copy of the GNU Lesser General Public
+  ~ License along with this software; if not, write to the Free
+  ~ Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<module xmlns="urn:jboss:module:1.5" name="org.jboss.as.product" slot="wildfly-ee">
```
Comment (yersan) [inline ee-galleon-pack/src/main/resources/packages/product.conf/content/modules/system/layers/base/org/jboss/as/product/wildfly-ee/module.xml]: Should not this change be also effective for the legacy product module.xml?
https://github.com/wildfly/wildfly/blob/master/dist-legacy/src/distribution/resources/modules/system/layers/base/org/jboss/as/product/wildfly-full/module.xml#L25

We could add and additional verification to ensure modules/system/layers/base/org/jboss/as/product/wildfly-ee/module.xml has the correct one

File: ee-galleon-pack/pom.xml [hunk 2]
```diff
@@ -0,0 +1,4155 @@
...
+
+        <dependency>
+            <groupId>org.eclipse.microprofile.rest.client</groupId>
+            <artifactId>microprofile-rest-client-api</artifactId>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>org.eclipse.microprofile.config</groupId>
+            <artifactId>microprofile-config-api<
```
[the remark this answers is not in the document]
Comment (bstansberry) [reply ee-galleon-pack/pom.xml]: Thanks; it looks like I missed deleting this one. I'll update the relevant commit.

File: ee-galleon-pack/src/main/resources/packages/product.conf/content/modules/system/layers/base/org/jboss/as/product/wildfly-ee/module.xml (hunk 1 above)
Comment (bstansberry) [reply ee-galleon-pack/src/main/resources/packages/product.conf/content/modules/system/layers/base/org/jboss/as/product/wildfly-ee/module.xml]: I think it should, yes.

Comment (bstansberry) [reply ee-galleon-pack/src/main/resources/packages/product.conf/content/modules/system/layers/base/org/jboss/as/product/wildfly-ee/module.xml]: Oh, definitely it should, as the name of the module dir was not matching the module.xml!

**Review by yersan — APPROVED** [review]

File: docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc [hunk 3]
```diff
@@ -115,6 +115,14 @@ WildFly defined layers to assemble a custom configuration. <<galleon-cli-recipes
 
 Basic layers:
 
+* _microprofile-fault-tolerance_: Support for MicroProfile Fault Tolerance.
+* _microprofile-jwt_: Support for MicroProfile JWT.
```
Comment (jfdenise) [inline docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc]: These layers are still documented in the WildFly EE layers, should be removed from EE.

File: feature-pack/pom.xml [hunk 4]
```diff
@@ -332,17 +332,6 @@
             <artifactId>jakarta.security.enterprise</artifactId>
         </dependency>
 
-        <dependency>
```
Comment (dosoudil) [inline feature-pack/pom.xml]: If these artifacts are removed from feature-pack, they're not runtime artifacts anymore.
Moving them to galleon-pack means that they're build dependencies, not runtime.
Is it correct from the perspective of using WildFly Server?
cc @rgdoliveira

File: docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc (hunk 3 above)
Comment (bstansberry) [reply docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc]: Thanks; will fix.

File: feature-pack/pom.xml (hunk 4 above)
Comment (bstansberry) [reply feature-pack/pom.xml]: They are still runtime dependencies of any server that uses the 'wildfly' galleon pack. The output of this maven module and the 'servlet-feature-pack' maven module are 'legacy feature packs' and are not used in any WildFly distribution. We just maintain these modules so projects that still want to create custom WF-technology-based servers using the legacy feature packs can use their output. But an official WildFly dist does not use the output of these. (You could do 'mvn install -pl dist -am' and this maven module wouldn't even build.) 

However, our official dists do use some *source* from these maven modules. This is so we can build the official Galleon feature packs and these legacy ones and not have to duplicate the source, which would almost certainly lead to bugs.  The Galleon feature pack poms actually copy over the source resources they need:

https://github.com/wildfly/wildfly/blob/master/galleon-pack/pom.xml#L73
https://github.com/wildfly/wildfly/blob/master/servlet-galleon-pack/pom.xml#L548

File: docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc (hunk 3 above)
Comment (bstansberry) [reply docs/src/main/asciidoc/_admin-guide/Galleon_provisioning.adoc]: Fixed.

**Comment by bstansberry** [comment]
retest this please

**Comment by bstansberry** [comment]
retest this please

**Pull request merged**